### PR TITLE
use the parse-key package to allow more complex keybindings

### PR DESCRIPTION
**Pull request by jlongster**
I chose this library because it's way, way smaller than mice and we don't actually need a full keybinding library. We already have React to handle events. We just need to parse strings, and library seems to work well and it tiny.

Tested locally and seems to work. You can now specify keys such as `ctrl-shift-H`.

### CODE REVIEW

File: src/DockMonitor.js [hunk 1]
```diff
@@ -44,24 +49,25 @@ export default class DockMonitor extends Component {
...
-    if (!e.ctrlKey) {
-      return;
-    }
+    const { visibilityKey, positionKey } = this.parsedKeys;
 
-    const key = e.keyCode || e.which;
-    const char = String.fromCharCode(key);
-    switch (char.toUpperCase()) {
-    case this.props.toggleVisibilityKey.toUpperCase():
+    if(this.matchesKey(visibilityKey, e)) {
```
Comment (gaearon) [inline src/DockMonitor.js]: Nit: mind adding a space after `if`?

File: src/DockMonitor.js [hunk 2]
```diff
@@ -44,24 +49,25 @@ export default class DockMonitor extends Component {
...
       e.preventDefault();
       this.props.dispatch(toggleVisibility());
-      break;
-    case this.props.changePositionKey.toUpperCase():
+    } else if(this.matchesKey(positionKey, e)) {
```
Comment (gaearon) [inline src/DockMonitor.js]: Same: a space.

File: src/DockMonitor.js [hunk 3]
```diff
@@ -34,6 +35,10 @@ export default class DockMonitor extends Component {
     super(props);
     this.handleKeyDown = this.handleKeyDown.bind(this);
     this.handleSizeChange = this.handleSizeChange.bind(this);
+    this.parsedKeys = {
```
Comment (gaearon) [inline src/DockMonitor.js]: Let's parse them in the event handler instead.
Feels wrong to remember something based on props.
They're not likely to change in our case, but still.

**Comment by gaearon** [comment]
Thanks, this looks good.
Please see the inline comments above for small changes I'd like to be made.

**Comment by jlongster** [comment]
Sure, if we have any perf problems with parsing the keys for each event we can figure it out later.

**Comment by gaearon** [comment]
:+1:

**Comment by benarmston** [comment]
I've just upgraded to using HEAD as I wanted to get the fix in 1e6980ae181bb773b0d5451d5316ad1b011789ec.  After the upgrade I ended up spending a few minutes debugging why my keybindings to show and hide the dock were no longer working.  After changing `toggleVisibilityKey="H"` to `toggleVisibilityKey="ctrl-h"` they started working again.

You're probably both aware of that change in behaviour, but I thought I'd mention it here to make sure that it gets mentioned in the release notes.  And in case anyone else comes looking at this PR trying to figure it out.

**Comment by gaearon** [comment]
Yeah, thanks for mentioning. Indeed, this was changed. I'll definitely mention it in the release notes when I get back to this project (likely in a week or two). Cheers!

**Pull request merged**